### ISSUE TITLE
ReferenceInput: accept function as linkType prop

### ISSUE THREAD

**Issue by jheiska** [marmelab/react-admin#2288]
Everything in my DB is related to a specific building. I have multiple tabs in BuildingShow-view showing different kinds of information, and I'd like to be able to link directly to the appropriate tab from the List views of these Resources.

**Describe the solution you'd like**
I'd like to be able to define the tab a ReferenceField links to. For example, from my "AlertsList" I could include a **linkTab="alerts"** -option:

```jsx
<ReferenceField
    label="building"
    reference="buildings"
    source="BuildingId"
    linkType="show"
    linkTab="alerts"
>
    <TextField source="BuildingName" />
</ReferenceField>
```

I could write my own Field component and define the link there. I just thought this would be a neat feature now that we have the possibility to link to specific tabs.

**Comment by fzaninotto** [comment]
In fact, it would be a good idea to accept a function as `linkType` prop, just like we accept a function as `redirect` prop in forms.

**Comment by fzaninotto** [comment]
Fixed by #3282
